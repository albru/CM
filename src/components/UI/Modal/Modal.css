.Modal {
    position: fixed;
    justify-content: center;
    z-index: 500;
    background-color: white;
    width: 85%;
    border: 1px solid #ccc;
    box-shadow: 1px 1px 1px black;
    padding: 16px;
    left: 7.5%;
    top: 20%;
    box-sizing: border-box;
    transition: all 0.3s ease-out;
    text-align: center;
    border-radius: 16px;
}

.CloseBtn {
    width: 20px;
    height: 20px;
    position: absolute;
    right: 20px;
    cursor: pointer;
}

.Stick1 {
    position: absolute;
    background-color: var(--color-lightgray);
    width: 20px;
    height: 3px;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) Rotate(230deg);

}

.Stick2 {
    position: absolute;
    background-color: var(--color-lightgray);
    width: 20px;
    height: 3px;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) Rotate(130deg);
}



@media (min-width: 600px) {
    .Modal {
        top: 30%;
        width: 500px;
        left: calc(50% - 250px);
    }
    .Stick1, .Stick2 {
        transition: all .3s;
    }
    .CloseBtn:hover .Stick1,
    .CloseBtn:hover .Stick2 {
        background-color: var(--color-second);
    }
}