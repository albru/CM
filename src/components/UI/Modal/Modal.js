import React from 'react';
import Aux from '../../../hoc/_Aux/_Aux';
import Backdrop from '../Backdrop/Backdrop';
import classes from './Modal.css';

const modal = props => {
    return (
        <Aux>
            <Backdrop isShow={props.show} close={props.close}/>
            <div className={classes.Modal}
                 style={{
                        transform: props.show ? 'translateY(0)' : 'translateY(-100vh)',
                        opacity: props.show ? '1' : '0'
                    }}>
                <div className={classes.CloseBtn}
                     onClick={props.close}>
                    <div className={classes.Stick1}></div>
                    <div className={classes.Stick2}></div>
                </div>
                {props.children}
            </div>
        </Aux>
    )
}

export default modal;