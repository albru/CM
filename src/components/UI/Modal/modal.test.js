import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Modal from './Modal';

configure({ adapter: new Adapter() })

describe('Modal', () => {
    it('should render correctly', () => {
        const output = shallow(
            <Modal />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});