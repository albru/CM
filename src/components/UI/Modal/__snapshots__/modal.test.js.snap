// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
<Aux>
  <Backdrop />
  <div
    style={
      Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      }
    }
  >
    <div
      onClick={[Function]}
    >
      <div />
      <div />
    </div>
  </div>
</Aux>
`;
